smarts/core/smarts.py

```diff
@@ -337,6 +370,7 @@ def reset(self, scenario: Scenario):
         self._total_sim_time += self._elapsed_sim_time
         self._elapsed_sim_time = 0
         self._step_count = 0
+        self._reset_required = False
```

Comment:
I might move this to the very end of this function, just in case something between here and then raises.

Reply:
My first instinct was to do that but it needs to be set `False` before calling `step` or gathering observations. If there is a raise again it will be set true again anyway and forces another `reset`. I had to make a specific case for where there is an exception in `reset` as you saw with retries since it is the responsibility of `reset` to provide a working simulation state.